common/util.go

```diff
@@ -92,7 +92,7 @@ func HandleHTTPRequest(message string, headers http.Header, data ConnectorMetada
 	}
 
 	if resp.StatusCode < 200 && resp.StatusCode > 300 {
-		return nil, fmt.Errorf("request returned failure: %v. http_endpoint: %v, source: %v", resp.StatusCode, data.HTTPEndpoint, data.SourceName)
```

Comment:
Added here so that we can pull the headers though (requirement to build the key). This can also be used to extract the response body.